.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - synchronize
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        ghc: ["8.0.2", "8.2.2", "8.4.4", "8.6.5", "8.8.3", "8.10.1"]
+        os: [ubuntu-latest]
+        include:
+          - os: macos-latest
+            ghc: "8.10.1"
+          - os: ubuntu-16.04
+            ghc: "7.10.3"
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Zsh
+        run: |
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt-get update
+            sudo apt-get install zsh
+          else
+            brew update
+            brew install zsh
+          fi
+      - uses: actions/setup-haskell@v1.1.4
+        id: setup-haskell-cabal
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: "3.0"
```

Comment:
@ibarrae any particular reason to pin the cabal version? Perhaps we could try the last one available instead, it seems to be the general recommendation.

![image](https://user-images.githubusercontent.com/2049686/118174241-2fc89a00-b3f4-11eb-962e-3bc7c038aa5f.png)

Source:
https://github.com/haskell/actions/tree/main/setup#version-support

Reply:
Nothing in particular, it just a copy-past of the old travis script, since we're on the plan to update the cabal version, maybe is also a good idea to use `v2-` commands too?